tools/envoy/fetch.sh

```diff
@@ -15,26 +16,33 @@ source "$(dirname -- "${BASH_SOURCE[0]}")/../common.sh"
 
 function download_envoy() {
     local binary_name=$1
+    local bin_dir=$(dirname ${BINARY_PATH})
     echo "Downloading ${binary_name}"
 
     if [ ! -d "$(dirname "${BINARY_PATH}")" ]; then
       mkdir -p "$(dirname "${BINARY_PATH}")"
     fi
 
+    local tar_path="${BINARY_PATH}.tar.gz"
+
     local status
-    status=$(curl -# --location --output "${BINARY_PATH}" --write-out '%{http_code}' \
-    "https://download.konghq.com/mesh-alpine/${binary_name}")
+    status=$(curl -# --location --output "${tar_path}" --write-out '%{http_code}' \
+    "https://github.com/kumahq/envoy-builds/releases/download/${ENVOY_TAG}/${binary_name}.tar.gz")
 
     if [ "$status" -ne "200" ]; then
         rm "${BINARY_PATH}"
         msg_err "Error: failed downloading Envoy: ${status} error"
     fi
 
+    tar -C "$bin_dir" -xzvf "${tar_path}"
+    rm $tar_path
+    mv "${bin_dir}/envoy-${ENVOY_DISTRO}" ${BINARY_PATH}
+
     [ -f "${BINARY_PATH}" ] && chmod +x "${BINARY_PATH}"
 }
 
-if [[ -n "${ENVOY_VERSION}" ]]; then
-  BINARY_NAME="envoy-${ENVOY_VERSION}-${ENVOY_DISTRO}-${GOARCH}"
+if [[ -n "${ENVOY_TAG}" ]]; then
+  BINARY_NAME="envoy-${GOOS}-${GOARCH}-${ENVOY_TAG}-${ENVOY_DISTRO}-opt${ENVOY_ARTIFACT_EXT}"
```

Comment:
if `${ENVOY_ARTIFACT_EXT}` is optional, maybe we should move `-opt` to it, as if there is no option, there will be a dangling `-opt` at the end of binary name - wdyt?

Reply:
Technically we could move `opt` to `ENVOY_ARTIFACT_EXT` however...

the "opt" part is an indication of whether the binary is "production" or "debug".
https://github.com/425devon/envoy-builds/blob/cd16b3e388ee8fa167b17b44c421226484d779f6/.github/workflows/ci_amd.yaml#L258
We always want to use "production" hence hardcoded "opt".
"opt" or "debug" is required.
artifact ext is optional.

Reply:
explained offline. Mutually agreed to close it